collect_app/src/test/java/org/odk/collect/android/projects/AddProjectDialogTest.kt

```diff
@@ -32,8 +39,22 @@ class AddProjectDialogTest {
         val scenario = RobolectricHelpers.launchDialogFragment(AddProjectDialog::class.java)
         scenario.onFragment { f: AddProjectDialog ->
             val dialog = f.dialog
-            Espresso.onView(ViewMatchers.isRoot()).perform(ViewActions.pressBack())
+            onView(isRoot()).perform(ViewActions.pressBack())
             assertThat(dialog!!.isShowing, `is`(false))
         }
     }
+
+    @Test
+    fun oneCharacterOrEmojiOnly_shouldBeAcceptedAsProjectIcon() {
+        val scenario = RobolectricHelpers.launchDialogFragmentInContainer(AddProjectDialog::class.java)
+        scenario.onFragment {
+            // One character
+            onView(withHint(R.string.project_icon)).perform(ViewActions.typeText("XYZ"))
+            onView(Matchers.allOf(withText("X"), withEffectiveVisibility(Visibility.VISIBLE))).check(matches(isDisplayed()))
```

Comment:
I think using `withHint` again here would make the test a little easier to read (it'd be clearer that the same view is being checked). I just checked and doing `onView(allOf(withHint(R.string.project_icon), withText("X")))` passes and fails as expected.

Reply:
I'm definitely liking that the one character and emoji case are covered here, but I'm thinking it might be nice to split them, so it's clear the one character part is testing that entering **more than one** character is truncated.

Reply:
Ok, done.